staging/src/kubevirt.io/api/clone/v1alpha1/types.go

```diff
@@ -0,0 +1,155 @@
+
+	Spec   VirtualMachineCloneSpec   `json:"spec" valid:"required"`
+	Status VirtualMachineCloneStatus `json:"status,omitempty"`
+}
+
+type CloneIncludeExclude struct {
+	// +optional
+	CloneAllByDefault *bool `json:"cloneAllByDefault,omitempty"`
+	// +optional
+	// +listType=atomic
+	Exclude []string `json:"exclude,omitempty"`
+	// +optional
+	// +listType=atomic
+	Include []string `json:"include,omitempty"`
+}
+
+type VirtualMachineCloneSpec struct {
+	Source *corev1.TypedLocalObjectReference `json:"source"`
+
+	// If the target is not provided, a random name would be generated for the target.
+	// The target's name can be viewed by inspecting status "TargetName" field below.
+	// +optional
+	Target *corev1.TypedLocalObjectReference `json:"target,omitempty"`
+
+	// +optional
+	Annotations *CloneIncludeExclude `json:"annotations,omitempty"`
+	// +optional
+	Labels *CloneIncludeExclude `json:"labels,omitempty"`
+
+	// NewMacAddresses manually sets that target interfaces' mac addresses. The key is the interface name and the
+	// value is the new mac address. If this field is not specified, a new MAC address will
+	// be generated automatically, as for any interface that is not included in this map.
+	// +optional
+	NewMacAddresses map[string]string `json:"newMacAddresses,omitempty"`
+	// NewSMBiosSerial manually sets that target's SMbios serial. If this field is not specified, a new serial will
+	// be generated automatically.
+	// +optional
+	NewSMBiosSerial *string `json:"newSMBiosSerial,omitempty"`
+}
+
+type VirtualMachineClonePhase string
+
+const (
+	PhaseUnset         VirtualMachineClonePhase = ""
+	SnapshotInProgress VirtualMachineClonePhase = "SnapshotInProgress"
+	CreatingTargetVM   VirtualMachineClonePhase = "CreatingTargetVM"
+	RestoreInProgress  VirtualMachineClonePhase = "RestoreInProgress"
+	Succeeded          VirtualMachineClonePhase = "Succeeded"
+	Failed             VirtualMachineClonePhase = "Failed"
+	Unknown            VirtualMachineClonePhase = "Unknown"
+)
+
+type VirtualMachineCloneStatus struct {
+	// +optional
+	// +nullable
+	CreationTime *metav1.Time `json:"creationTime,omitempty"`
+
+	// +optional
+	Phase VirtualMachineClonePhase `json:"phase,omitempty"`
+
+	// +optional
+	// +listType=atomic
+	Conditions []Condition `json:"conditions,omitempty"`
+
+	// +optional
+	// +nullable
+	SnapshotName *string `json:"snapshotName,omitempty"`
+
+	// +optional
+	// +nullable
+	RestoreName *string `json:"restoreName,omitempty"`
+
+	// +optional
+	// +nullable
+	TargetName *string `json:"targetName,omitempty"`
+}
+
+// ConditionType is the const type for Conditions
+type ConditionType string
+
+const (
+	// ConditionReady is the "ready" condition type
+	ConditionReady ConditionType = "Ready"
+
+	// ConditionProgressing is the "progressing" condition type
+	ConditionProgressing ConditionType = "Progressing"
+
+	// ConditionFailure is the "failure" condition type
+	ConditionFailure ConditionType = "Failure"
```

Comment:
Not sure we need progressing and failure?

Reply:
I'm not opposed to error but I think ready and progressing are sufficient.  Error can kind of be inferred when progressing is false